src/BaseQuery.php

```diff
@@ -646,13 +669,30 @@ protected function innerJoin($join, $alias, $conditionType = null, $condition =
 
 	private function commonJoin($joinType, $join, $alias, $conditionType = null, $condition = null, $indexBy = null): self {
 		$join = $this->addColumnPrefix($join);
-		$filterKey = implode('_', [$joinType, $join, $alias, (string)$conditionType, (string)$condition, (string)$indexBy]);
+		$filterKey = $this->getJoinFilterKey($joinType, $join, $alias, $conditionType, $condition, $indexBy);
 		$this->filter[$filterKey] = function (QueryBuilder $qb) use ($joinType, $join, $alias, $conditionType, $condition, $indexBy) {
 			$qb->$joinType($join, $alias, (string)$conditionType, (string)$condition, (string)$indexBy);
 		};
 		return $this;
 	}
 
+	private function getJoinFilterKey($joinType, $join, $alias, $conditionType = null, $condition = null, $indexBy = null) {
+		return implode('_', [$joinType, $join, $alias, (string)$conditionType, (string)$condition, (string)$indexBy]);
+	}
+
+	private function isJoinFilterExisted($filterKey, bool $withType = true) {
+		if (!$withType) {
```

Comment:
fakt tento parametr potrbujeme?

Reply:
Tady je pravda, že buď ho vynechat, nebo dát tuhle metodu protected. Jsem pro protected, kdyby to někdo chtěl využít ve svém Query.

Reply:
jako ze by to nekdo chtel innerJoin a podruhe leftJoin?

Reply:
ano